File: finance-ui/src/services/invoiceService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/v1/invoices';

export const createInvoice = (invoiceData) =>
  axios.post(API_URL, invoiceData);

export const getInvoices = () =>
  axios.get(API_URL);

export const getInvoiceById = (id) =>
  axios.get(`${API_URL}/${id}`);

export const getInvoicesByContractId = (contractId) =>
  axios.get(`${API_URL}/contract/${contractId}`);

export const getInvoicesByOrganizationId = (organizationId) =>
  axios.get(`${API_URL}/organization/${organizationId}`);

export const searchInvoices = (query) =>
  axios.get(`${API_URL}/search`, {
    params: { query },
  });

export const updateInvoice = (id, invoiceData) =>
  axios.put(`${API_URL}/${id}`, invoiceData);

export const deleteInvoice = (id) =>
  axios.delete(`${API_URL}/${id}`);

File: src/main/java/com/xebia/finance/invoice/service/InvoiceService.java
package com.xebia.finance.invoice.service;

import com.xebia.finance.invoice.dto.CreateInvoiceRequest;
import com.xebia.finance.invoice.dto.InvoiceResponse;
import com.xebia.finance.invoice.dto.UpdateInvoiceRequest;

import java.util.List;
import java.util.UUID;

/**
 * Service interface for invoice operations.
 *
 * @author Pratik Kumar | Xebia Virtual Internship | 70812345pratik@gmail.com
 */
public interface InvoiceService {

    /**
     * Create a new invoice.
     */
    InvoiceResponse createInvoice(CreateInvoiceRequest request);

    /**
     * Get all invoices.
     */
    List<InvoiceResponse> getAllInvoices();

    /**
     * Get invoice by ID.
     */
    InvoiceResponse getInvoiceById(UUID id);

    /**
     * Get invoices by contract ID.
     */
    List<InvoiceResponse> getInvoicesByContractId(UUID contractId);

    /**
     * Get invoices by organization ID.
     */
    List<InvoiceResponse> getInvoicesByOrganizationId(UUID organizationId);

    /**
     * Update an existing invoice.
     */
    InvoiceResponse updateInvoice(UUID id, UpdateInvoiceRequest request);

    /**
     * Delete an invoice.
     */
    void deleteInvoice(UUID id);

    /**
     * Search invoices by invoice number, organization name, or other criteria.
     */
    List<InvoiceResponse> searchInvoices(String query);
}
